Sources/StreamChat/WebSocketClient/Events/MessageEvents.swift

```diff
@@ -4,78 +4,168 @@
+            userId = try response.value(at: \.user?.id)
+            messageId = try response.value(at: \.message?.id)
+            createdAt = try response.value(at: \.createdAt)
+            payload = response
+        }
+    }
     
-    init(from response: EventPayload) throws {
-        userId = try response.value(at: \.user?.id)
-        cid = try response.value(at: \.cid)
-        messageId = try response.value(at: \.message?.id)
-        updatedAt = try response.value(at: \.message?.updatedAt)
-        payload = response
+    init?(event: DTO, session: DatabaseSession) {
+        guard
+            let user = session.user(id: event.userId),
+            let message = session.message(id: event.messageId)
+        else { return nil }
+        
+        self.init(
+            cid: event.cid,
+            user: user.asModel(),
+            message: message.asModel(),
+            createdAt: event.createdAt
+        )
     }
 }
 
-public struct MessageDeletedEvent: MessageSpecificEvent {
-    public let userId: UserId
+public struct MessageDeletedEvent: Event {
     public let cid: ChannelId
-    public let messageId: MessageId
-    public let deletedAt: Date
-    
-    let payload: Any
+    public let user: ChatUser
+    public let message: ChatMessage
+    public let createdAt: Date
+}
+
+extension MessageDeletedEvent {
+    struct DTO: MessageSpecificEvent {
+        let cid: ChannelId
+        let messageId: MessageId
+        let userId: UserId
+        let createdAt: Date
+        let payload: Any
+        
+        init(from response: EventPayload) throws {
+            cid = try response.value(at: \.cid)
+            messageId = try response.value(at: \.message?.id)
+            userId = try response.value(at: \.user?.id)
+            createdAt = try response.value(at: \.createdAt)
+            payload = response
+        }
+    }
     
-    init(from response: EventPayload) throws {
-        userId = try response.value(at: \.user?.id)
-        cid = try response.value(at: \.cid)
-        messageId = try response.value(at: \.message?.id)
-        deletedAt = try response.value(at: \.message?.deletedAt)
-        payload = response
+    init?(event: DTO, session: DatabaseSession) {
+        guard
+            let user = session.user(id: event.userId),
+            let message = session.message(id: event.messageId)
+        else { return nil }
+        
+        self.init(
+            cid: event.cid,
+            user: user.asModel(),
+            message: message.asModel(),
+            createdAt: event.createdAt
+        )
     }
 }
 
 /// `ChannelReadEvent`, this event tells that User has mark read all messages in channel.
 public typealias ChannelReadEvent = MessageReadEvent
 
 /// `ChannelReadEvent`, this event tells that User has mark read all messages in channel.
-public struct MessageReadEvent: UserSpecificEvent, ChannelSpecificEvent {
-    public let userId: UserId
+public struct MessageReadEvent: Event {
```

Comment:
There's no `messageId` in the event so having names vice versa would make more sense: `ChannelReadEvent` - a real type and `MessageReadEvent` as a second name